packages/base/src/features/F6Navigation.ts

```diff
@@ -172,8 +172,27 @@ class F6Navigation {
 	}
 
 	updateGroups() {
+		const container = this.findContainer();
+
 		this.setSelectedGroup();
-		this.groups = getFastNavigationGroups(document.body);
+		this.groups = getFastNavigationGroups(container);
+	}
+
+	findContainer() {
+		const htmlElement = window.document.querySelector("html");
+		let element = this.deepActive(window.document);
+
+		while (element && element !== htmlElement) {
+			const closestScopeEl = element.closest<HTMLElement>("[data-sap-ui-fastnavgroup-container='true']");
```

Comment:
or maybe just

`element.closest<HTMLElement>("[data-sap-ui-fastnavgroup-container]");`

because IMO that's also valid. Or is the contract to specifically set to `true` ?

Reply:
As we discussed, it is made to be consistent with other F6 data-* attributes and how they work.